python/pyspark/sql/dataframe.py

```diff
@@ -674,7 +674,7 @@ def cache(self):
         .. note:: The default storage level has changed to `MEMORY_AND_DISK` to match Scala in 2.0.
         """
         self.is_cached = True
-        self._jdf.cache()
+        self.persist(StorageLevel.MEMORY_AND_DISK)
```

Comment:
Hm, this should already be the default on the Scala side. I am not clear why this changes behavior.
I understand the idea but also not sure of all the implications here as so much has changed since SPARK-2014

Reply:
On scala side, default is `StorageLevel(true, true, false, true)` which has `deserialized=true`. It is getting used when scala `cache()` method is called directly as in the case with the current master branch.

However, since the data is already serialized on the Python side, Storage level to be used should have `deserialized=false`. With this code change persist will be called with `StorageLevel(True, True, False, False)` and hence deserialized will be set to false.

Reply:
I see, python StorageLevels are defined differently. Erm, @ScrapCodes or say @cloud-fan would you have an opinion on this? it does seem plausible but I don't know if there's a hidden reason for this difference.

Reply:
For SPARK-2014, I cannot remember everything 6 years back. While I scrounge my memory, a bit more.

In python, turning on serialised should be okay, as everything is already serialized anyway.